apis/docdb/v1alpha1/custom_types.go

```diff
@@ -156,6 +157,81 @@ type CustomDBClusterParameters struct {
 	VPCSecurityGroupIDsRefs []xpv1.Reference `json:"vpcSecurityGroupIDsRefs,omitempty"`
 	// TODO(haarchri): when resource is bumped to beta we will convert this field to vpcSecurityGroupIdSelector
 	VPCSecurityGroupIDsSelector *xpv1.Selector `json:"vpcSecurityGroupIDsSelector,omitempty"`
+
+	// RestoreFrom specifies the details of the backup to restore when creating a new DBCluster.
+	// +optional
+	RestoreFrom *RestoreDBClusterBackupConfiguration `json:"restoreFrom,omitempty"`
+}
+
+// SnapshotRestoreBackupConfiguration defines the details of the snapshot to restore from.
+type SnapshotRestoreBackupConfiguration struct {
+	// SnapshotIdentifier is the identifier of the snapshot to restore.
+	SnapshotIdentifier *string `json:"snapshotIdentifier"`
+}
+
+// PointInTimeRestoreDBClusterBackupConfiguration defines the details of the time to restore from
+type PointInTimeRestoreDBClusterBackupConfiguration struct {
+	// RestoreTime is the date and time (UTC) to restore from.
+	// Must be before the latest restorable time for the DB instance.
+	// Can't be specified if the useLatestRestorableTime parameter is enabled.
+	// Example: 2011-09-07T23:45:00Z
+	// +optional
+	RestoreTime *metav1.Time `json:"restoreTime"`
```

Comment:
Required fields should not be a pointer. Optional fields should be a pointer and have the `omitempty` setting.